src/averageAges.js

```diff
@@ -55,7 +68,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  let kids = people.filter(kid => people.some(person =>
+    kid.mother === person.name));
+
+  kids = onlyWithSon
+    ? kids.filter(kid => kid.sex === 'm')
+    : kids;
+
+  const mothers = kids.map(kid =>
+    people.find(mother => mother.name === kid.mother));
+  const ageDifference = kids.map((kid, index) =>
+    kid.born - mothers[index].born);
+  const totalDifference = ageDifference.reduce((sum, age) => sum + age, 0);
```

Comment:
Instead of having two `map` loops, move logic inside the `reduce` call. It will be one array loop instead of 3.